### PR TITLE
Improve user interface

### PR DESCRIPTION
**Pull request by ofekashery**
## What's Changed?
- Restructure the workspace folders
- Remove the ``auth.expo.io`` usage **(needs Android testing)**
- Redesign pages
- Use Heebo font
- FIx iOS safe area

<hr />

### CODE REVIEW

File: components/header.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { Image, View } from 'react-native'
+import { Appbar } from 'react-native-paper'
+import FIRSTLogo from '../assets/images/FIRST-logo.png'
```
Comment (idanstark42) [inline components/header.js]: Keep a whitespace between external and internal imports

File: components/page-template.js [hunk 2]
```diff
@@ -1,13 +1,11 @@
 import React, { useState, useEffect } from 'react'
-import { View, Text, Dimensions, Image } from 'react-native'
+import { View, Text, Dimensions, Image, SafeAreaView } from 'react-native'
 import { RFValue } from 'react-native-responsive-fontsize'
 import * as ScreenOrientation from 'expo-screen-orientation'
-import { Appbar, Button } from 'react-native-paper'
+import { Button } from 'react-native-paper'
+import Header from './header'
```
Comment (idanstark42) [inline components/page-template.js]: Keep a whitespace between external and internal imports

File: components/team-item.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { TouchableOpacity } from 'react-native-gesture-handler'
+import { List } from 'react-native-paper'
+import i18n from '../lib/i18n'
```
Comment (idanstark42) [inline components/team-item.js]: Keep a whitespace between external and internal imports

File: components/team-item.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import moment from '../lib/moment'
+
+const programColors = {
```
Comment (idanstark42) [inline components/team-item.js]: Move this to config

File: components/team-item.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  'FIRST LEGO League Discover': '#662D91',
+  'FIRST LEGO League Explore': '#00A651',
+  'FIRST LEGO League Challenge': '#ED1C24',
+  'FIRST Tech Challenge': '#F57E25',
+  'FIRST Robotics Competition': '#009CD7'
+}
+
+const formatDates = event => {
```
Comment (idanstark42) [inline components/team-item.js]: Move this to `lib` too. It's not the responsibility of this component

File: components/team-item.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            {...props}
+            icon="account-multiple-outline"
+            color={programColors[team.program]}
+            style={{
+              backgroundColor: programColors[team.program] + '1f',
+              borderRadius: 8
+            }}
+          />
+        )}
+        {...props}
```
Comment (idanstark42) [inline components/team-item.js]: Why keep unused props?

File: components/team-item.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    return start.format(i18n.t('datetime_formats.long', { endTime }))
+  } else {
+    const date1 = start.format(i18n.t('datetime_formats.short'))
+    const date2 = end.format(i18n.t('datetime_formats.short'))
+    return i18n.t('datetime_formats.2dates', { date1, date2 })
+  }
+}
+
+const TeamItem = ({ item, onPress, ...props }) => {
+  const { team } = item
```
Comment (idanstark42) [inline components/team-item.js]: This funciton should already reviece the `team` as a prop. It's not its responsibility to extract it from `item`. The logic for getting the teams from the events is in the `home` screen, and should not be split into seperate places.

File: lang/en.js [hunk 8]
```diff
@@ -1,21 +1,40 @@
+import React from 'react'
+import { Text } from 'react-native-paper'
```
Comment (idanstark42) [inline lang/en.js]: The `lang` files should remain decoupled from the platform in which we work, thereafore they shouldn't import `react` or `react-native` and shuoldn't use elements from there. If you must, create a logic at whoever uses those texts to add the elements in there.

File: lang/en.js [hunk 9]
```diff
@@ -1,21 +1,40 @@
...
   shoot_another: 'לחצו כאן לצילום סרטון נוסף',
   exit: 'לא, תודה. סיימנו לעכשיו',
-  thank_you: 'תודה לכם על השתתפות בתחרות FLL Replay!'
+  thank_you: 'תודה לכם על השתתפות בתחרות FLL Replay!',
+
+  team_name: 'Team #%{number} - %{affiliation}',
+  datetime_formats: {
+    short: 'D בMMM בשעה HH:mm',
+    long: 'יום dd׳, D בMMMM, בשעה HH:mm עד %{endTime}',
+    '2dates': '%{date1} עד ה־%{date2}'
```
Comment (idanstark42) [inline lang/en.js]: The formal name with this is `datespan`

File: screens/home.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
+import React from 'react'
```
Comment (idanstark42) [inline screens/home.js]: Screens are components. If you want to seperate them into two groups, create the following folder tree:
 - components
    - screens
    - containers (also can be names 'generics', these are "dumb" components, with 
       small amount of logic)

File: lib/auth.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+          .split('&')
+          .map(i => i.split('=').map(i => decodeURIComponent(i))) ?? []
+      )
+
+      if (String(resParams.id_token) !== 'undefined' && String(resParams.id_token) !== 'null') {
+        return resParams.id_token || null
+      }
+    }
+  } catch (error) {
+    console.error(error)
```
Comment (idanstark42) [inline lib/auth.js]: Don't console.log here, throw the error to the UI and let the UI decide weather to show it to the user or not.

File: lib/auth.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import Constants from 'expo-constants'
+import * as WebBrowser from 'expo-web-browser'
+import jwtDecode from 'jwt-decode'
+import i18n from '../lib/i18n'
+
+import config from '../config'
+
+export const login = async () => {
```
Comment (idanstark42) [inline lib/auth.js:16]: Can you explain the change you did here?
Also this function is way to long. Make it shorter by creating some helper functions. For instance, why are you handling the splitting of the search parameters, when there are tons of libraries that can do it for you?

File: screens/post-match.js [hunk 13]
```diff
@@ -1,10 +1,10 @@
 import React from 'react'
 import { BackHandler } from 'react-native'
 
-import BasicPage from './basic_page'
-import i18n from '../logic/i18n'
+import BasicPage from '../components/page-template'
```
Comment (idanstark42) [inline screens/post-match.js]: Rename this to `PageTemplate` - keep consistent naming.

File: screens/login.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+        <Text style={styles.headline}>{i18n.t('welcome')}</Text>
+        <Text
+          style={{
+            fontSize: RFValue(16),
+            textAlign: 'center',
+            color: '#888',
+            marginTop: RFValue(12)
+          }}
+        >
+          אנחנו כבר לא יכולים לחכות לראות את הדברים עליהם עבדתם בעונה זו!
```
Comment (idanstark42) [inline screens/login.js]: Use `i18n` here

File: screens/login.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+import { View } from 'react-native'
+import { Button, Text } from 'react-native-paper'
+import { RFValue } from 'react-native-responsive-fontsize'
+
+import i18n from '../lib/i18n'
+import { login } from '../lib/auth'
+import BasicPage from '../components/page-template'
+
+export default function LoginScreen({ navigation }) {
+  const handleLogin = () => {
```
Comment (idanstark42) [inline screens/login.js]: `handleLogin` doesn't sound like it does the login process.

File: screens/login.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+    login().then(authToken => {
+      if (authToken) navigation.replace('HOME', { authToken })
+    })
+  }
+
+  return (
+    <BasicPage>
+      <View
+        style={{
```
Comment (idanstark42) [inline screens/login.js]: Keep the styles outside the `Component`

File: screens/login.js [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+          <Text style={{ fontSize: RFValue(14), color: '#fff', fontFamily: 'Heebo_700Bold' }}>
+            {i18n.t('login')}
+          </Text>
+        </Button>
+      </View>
+    </BasicPage>
+  )
+}
+
+const styles = {
```
Comment (idanstark42) [inline screens/login.js]: Move the styles here

File: screens/pre-instructor.js [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+      Dimensions.removeEventListener('change', checkLandscape)
+      ScreenOrientation.unlockAsync().catch(() => {})
+    }
+  }, [checkLandscape])
+
+  const handleStart = () => navigation.navigate('INST', { item: route.params.item })
+  const handleExit = () => navigation.pop()
+
+  if (!isLandscape) {
+    return (
```
Comment (idanstark42) [inline screens/pre-instructor.js]: This is a large component, I would move it to it's own component file.

File: screens/login.js [hunk 19]
```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { View } from 'react-native'
+import { Button, Text } from 'react-native-paper'
+import { RFValue } from 'react-native-responsive-fontsize'
+
+import i18n from '../lib/i18n'
+import { login } from '../lib/auth'
+import BasicPage from '../components/page-template'
```
Comment (idanstark42) [inline screens/login.js]: Use consistent naming. Use `PageTemplate` here.

File: screens/home.js [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+        </View>
+      ) : (
+        <>
+          <View style={styles.header}>
+            <Text style={styles.user}>
+              {i18n.t('hello_user', { name: getUserGivenName(authToken) })}
+            </Text>
+          </View>
+          <ScrollView style={{ width }}>
+            {Object.keys(events).map(event => (
```
Comment (idanstark42) [inline screens/home.js]: The 'key' of the object is not the event, it's the `eventKey` or the `eventId` or something. Maybe use `Object.entries` instead.

File: screens/home.js [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+              <View key={event}>
+                <List.Subheader>{event}</List.Subheader>
+                {events[event].map(item => (
+                  <TeamItem key={item.id} item={item} onPress={() => handleSelectTeam(item)} />
```
Comment (idanstark42) [inline screens/home.js]: `handleTeamSelect`

**Review by idanstark42 — CHANGES_REQUESTED** [review]
There are a few things I need you to change and explaing to me

File: lib/auth.js (hunk 12 above)
Comment (ofekashery) [reply lib/auth.js:16]: I reduced the function now.
The ``WebBrowser.openAuthSessionAsync`` removes the need to use ``auth.expo.io`` and gives us better control. The token reaches the ``appRedirect`` (our deep link), then the browser closes and we get the ``id_token`` by parsing the ``result.url``'s query parameters.

File: components/team-item.js (hunk 7 above)
Comment (ofekashery) [reply components/team-item.js]: Take a look at the updated code, we also use the ``event`` to render the dates.
https://github.com/FIRSTIsrael/remote-matches-app/blob/0bd2b3ace86abb5a61a71bab8bfb5f2724210bb7/components/team-item.js#L9-L15

File: screens/home.js [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+    })
+  }, [])
+
+  const width = Dimensions.get('window').width
+
+  const handleTeamSelect = item => navigation.navigate('PRE_INST', { item, authToken })
+
+  return (
+    <PageTemplate>
+      {!events ? (
```
Comment (idanstark42) [inline screens/home.js]: Notice the difference in the way you write the conditions. Here you write it within the jsx using the `condition ? onTrue : onFalse` syntax, but in `pre-instructor.js`  you write it as an `if () { } else { }`.
This is not something that will prevent the merge, but you should try to keep the language style consistent.

**Review by idanstark42 — CHANGES_REQUESTED** [review]
Very good and quick fixes :)!
The only ones left is the decoupling of the language files from the framework, and the structure of the components directory tree.

File: screens/home.js (hunk 22 above)
Comment (ofekashery) [reply screens/home.js]: In ``PreInstructorScreen``, the root component is different between the 2 states (``<PageTemplate />`` vs ``<PageTemplate hideHeader>``), while in ``HomeScreen``, the root component is the same.

**Pull request merged**